src/laser/MultiLaser.cpp

```diff
@@ -1139,10 +824,35 @@ MultiLaser::InitLaserSlice (const int islice, const int comp)
     for ( amrex::MFIter mfi(m_slices, DfltMfiTlng); mfi.isValid(); ++mfi ){
         const amrex::Box& bx = mfi.tilebox();
         amrex::Array4<amrex::Real> const & arr = m_slices.array(mfi);
-        // Initialize a Gaussian laser envelope on slice islice
-        //check point
-        for (int ilaser=0; ilaser < m_nlasers; ilaser++) {
+        // Initialize a laser envelope on slice islice
+        for (int ilaser = 0; ilaser < m_nlasers; ilaser++) {
             auto& laser = m_all_lasers[ilaser];
+            if (laser.m_laser_init_type == "from_file"){
+                amrex::Array4<amrex::Real> const& arr_ff = laser.m_F_input_file.array();
+                amrex::ParallelFor(
+                bx,
+                [=] AMREX_GPU_DEVICE(int i, int j, int k)
+                {
+                    if (ilaser == 0) {
+                        arr(i, j, k, comp ) = 0._rt;
+                        arr(i, j, k, comp + 1 ) = 0._rt;
+                    }
+                    arr(i, j, k, comp ) += arr_ff(i, j, k, 0 );
+                    arr(i, j, k, comp + 1 ) += arr_ff(i, j, k, 1 );
```

Comment:
```suggestion
                    arr(i, j, k, comp ) += arr_ff(i, j, islice, 0 );
                    arr(i, j, k, comp + 1 ) += arr_ff(i, j, islice, 1 );
```
I think the z index has to be modified here, likely should be islice.